docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+- `message.im`
+- `message.channels`
+
+![Subscribe to bot events](./images/subscribe-to-bot-events.png)
+
+Once you've done this, click "Save Changes"
+
+### Set OAuth & Permissions
+
+Under Features in the navigation bar, go to OAuth & Permissions.
+
+You'll need to enable the following Bot Token Scopes:
+
+- `channels:history`
+- `channels:read`
+- `chat:write`
+- `groups:history`
+- `groups:read`
+- `im:history`
+- `mpim:history`
+
+### Install App
+
+After you've done this, install the app in Slack by going to Install App under Settings in the navigation bar.
+
+### Get credentials
+
+You will need **Signing Secret** and **App Level Token** from the Basic Information page.
+And you will need the **Bot User OAuth Token** from the Install App page, which is also available on the OAuth & Permissions page.
+
+## Create TypeScript projects for the Slack App
+
+You need two TypeScript projects for this Slack application.
+You may have already created the first one for the Temporal Application in the [Prerequisite steps](/getting_started/typescript/dev_environment/).
+Here, make sure you have the second one for the Slack bot logic.
+
+This tutorial assumes that your TypeScript projects will be organized like this:
+
+```plaintext
+-- your-workqueue-slack-app
+  |-- temporal-application # This is the Temporal Application project
+  |   |-- .env
+  |   |-- src
+  |   |   |-- workflows
+  |   |   |   |-- workqueue.ts
+  |   |   |-- index.ts
+  |   |-- package.json
+  |   |-- tsconfig.json
+  |-- bot # This is the Slack bot project
+  |   |-- .env
+  |   |-- src
+  |   |   |-- index.ts
+  |   |-- package.json
+  |   |-- tsconfig.json
+  |-- common-types # These types are shared by both projects
+      |-- types.ts
+```
+
+This tutorial uses `npm` as the package manager.
+
+```
+npm init
+```
+
+And ensures there is a `tsconfig.json` file in each project root.
+
+```
+tsc --init
+```
+
+### Dependencies for the projects
+
+**Temporal Application dependencies**
+
+In the `temporal-application` project, you will need the following dependencies in your `package.json`:
+
+```json
+{
+  // ...
+  "dependencies": {
+    "@temporalio/worker": "^1.10.1",
+    "dotenv": "^16.1.3",
+    "path": "^0.12.7"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.7",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  }
+  // ...
+}
+```
+
+**Slack bot dependencies**
+
+In the `bot` project, you will need the following dependencies in your `package.json`:
+
+```json
+{
+  // ...
+  "dependencies": {
+    "@slack/bolt": "^3.19.0",
+    "@slack/web-api": "^6.12.0",
+    "@temporalio/client": "^1.7.4",
+    "crypto": "^1.0.1",
+    "date-fns": "^3.6.0",
+    "dotenv": "^16.0.3"
+  },
+  "devDependencies": {
+    "@types/node": "^20.14.7",
+    "ts-node": "^10.9.2",
+    "typescript": "^5.4.5"
+  }
+  // ...
+}
+```
+
+### Environment variables
+
+First, make sure you have `.env` added to your `.gitignore` file to avoid committing your credentials.
```

Comment:
Need another sentence here (or rearrange this section slightly) so it's introduced better -- "Your application will be using your Slack tokens to authenticate. You should not commit these to your project repository. Instead, store them in an `.env` file in your project root, and add that `.env` file to a `.gitignore` file in the project root so that they will not be accidentally committed.

Reply:
Nice.